doc/start/hardware-recommendations.rst

```diff
@@ -54,55 +54,60 @@ Monitors and managers (ceph-mon and ceph-mgr)
 option.
 
-- Setting the osd_memory_target below 2GB is typically not recommended (it may
-  fail to keep the memory that low and may also cause extremely slow performance.
+- Setting the :confval:`osd_memory_target` below 2GB is typically not
+  recommended (it may fail to keep the memory that low and may also cause
+  extremely slow performance.
 
 - Setting the memory target between 2GB and 4GB typically works but may result
-  in degraded performance as metadata may be read from disk during IO unless the
+  in degraded performance: metadata may be read from disk during IO unless the
   active data set is relatively small.
 
-- 4GB is the current default osd_memory_target size and was set that way to try
-  and balance memory requirements and OSD performance for typical use cases.
+- 4GB is the current default :confval:`osd_memory_target` size.  This default
+  was chosen for typical use cases, and is intended to balance memory
+  requirements and OSD performance for typical use cases.
 
-- Setting the osd_memory_target higher than 4GB may improve performance when
-  there are many (small) objects or large (256GB/OSD or more) data sets being
-  processed.
+- Setting the :confval:`osd_memory_target` higher than 4GB can improve
+  performance when there many (small) objects or large (256GB/OSD or more) data
+  sets are processed.
 
-.. important:: The OSD memory autotuning is "best effort".  While the OSD may
+.. important:: OSD memory autotuning is "best effort". Although the OSD may
    unmap memory to allow the kernel to reclaim it, there is no guarantee that
-   the kernel will actually reclaim freed memory within any specific time
-   frame.  This is especially true in older versions of Ceph where transparent
-   huge pages can prevent the kernel from reclaiming memory freed from
+   the kernel will actually reclaim freed memory within a specific time
+   frame. This is especially true in older versions of Ceph where transparent
+   huge pages can prevent the kernel from reclaiming memory that was freed from
    fragmented huge pages. Modern versions of Ceph disable transparent huge
-   pages at the application level to avoid this, though that still does not
-   guarantee that the kernel will immediately reclaim unmapped memory.  The OSD
-   may still at times exceed it's memory target.  We recommend budgeting around
-   20% extra memory on your system to prevent OSDs from going OOM during
-   temporary spikes or due to any delay in reclaiming freed pages by the
-   kernel.  That value may be more or less than needed depending on the exact
-   configuration of the system.
+   pages at the application level to avoid this, but that does not
+   guarantee that the kernel will immediately reclaim unmapped memory. The OSD
+   may still at times exceed its memory target. We recommend budgeting 
+   approximately 20% extra memory on your system to prevent OSDs from going 
```

Comment:
s/extra memory/RAM headroom/ ?
I personally favor 2x `osd_memory_target` but I don't know that I would go so far as to codify that.
v Below, how widely known is `OOM` ?  I hadn't heard of the oomkiller until 2014, maybe
s/OOM/OOM (Out Of Memory)/ ?   Yes, I know, you just want to merge the darn thing and move on ;)

Reply:
> Yes, I know, you just want to merge the darn thing and move on ;)

Yeah, but this kind of back-and-forth is, I must admit, very productive. It is true that I had hoped to get this closed today, but now that we've really dug into this page, I keep finding things that cry out for improvement.

Since I already have an approval, I'll make a few more changes and then merge this PR (so that the backporting can be done... learning to backport docs is a whole nother can of worms not psychologically dissimilar from learning to make a properly-formatted Ceph pull request, but I think that I've finally broken it and I'm curious to test that thought). Anything that we've discussed here that hasn't made it into the documentation will go into a hardware-recs-March-2022 PR.

Regarding the question of "OOM": I consider myself a fairly ignorant person, and I know that "OOM" means "out of memory", so I assume that it's common knowledge. But, as has been pointed out to me on more than one occasion, when it comes to assessing the limits of common knowledge, I am a fool. Anyway, I think s/OOM/OOM (Out Of Memory)/ or maybe s/OOM/(**O**ut **O**f **M**emory)/ is a good idea, because it provides the reader with information that he or she might not have had.

Regarding "extra memory vs RAM headroom": I don't know that there's a way that expresses this without raising even more questions. I can imagine someone wondering "is 'headroom' a term with a specific definition?". And I can imagine someone thinking "Is 'extra memory' supposed to mean 'more memory than I think I would need? And if it is, isn't it the case that by realizing that I need that extra amount of memory then the first amount of memory is wrong and the second amount of memory is right, and the second amount of memory is the amount that I should base the twenty percent on?". So on this recondite and picky-even-for-us matter, I'm going to kick the can down the road and revisit it in the aforementioned March 2022 hardware-recs PR.